libtransmission/inout.cc

```diff
@@ -271,7 +271,7 @@ static std::optional<tr_sha1_digest_t> recalculateHash(tr_torrent* tor, tr_piece
         if (auto const success = tr_cacheReadBlock(tor->session->cache, tor, piece, offset, len, std::data(buffer)) == 0;
             !success)
         {
-            tr_sha1_final(sha, nullptr);
```

Comment:
Hmm I hear what you're saying, but this would replace the current ambiguity with a new one, i.e. "do I need to call tr_sha1_free() after tr_sha1_final()".

I really like the idea of having one `[nodiscard]` but on balance it might be better to leave it as it is, since (nearly) all callers are using the `tr_sha1()` wrapper anyway?

Reply:
Some type of RAII would be nice I suppose, so that destruction is implicit and separate from hash finalization. Agreed to leave as is for the time being.